### PR TITLE
Add incidents field

### PR DESCRIPTION
**Pull request by wolyslager**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
Adding a rawJSON field to the incidents generated from the fetch_incidents command. 

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Review by sapirshuker — COMMENTED** [review]
Hi @wolyslager, Thank you for your contribution! Good work :)
We're ready for a demo. Please check [this page](https://xsoar.pan.dev/docs/contributing/demo-prep), and let me know when you're available for one over DFIR. 
Please fix the validation errors, you can see [here](https://app.circleci.com/pipelines/github/demisto/content/312108/workflows/60b1d55b-8e2d-43e1-9f35-0b26a8aa592c/jobs/726044)
Please feel free to reach out to me with any questions - I'm available here or on slack :)

**Comment by wolyslager** [comment]
> Hi @wolyslager, Thank you for your contribution! Good work :) We're ready for a demo. Please check [this page](https://xsoar.pan.dev/docs/contributing/demo-prep), and let me know when you're available for one over DFIR. Please fix the validation errors, you can see [here](https://app.circleci.com/pipelines/github/demisto/content/312108/workflows/60b1d55b-8e2d-43e1-9f35-0b26a8aa592c/jobs/726044) Feel free to reach out to me with any questions - I'm available here or on slack :)

Hey @sapirshuker! Thanks for taking a look 🙏  - I was unable to join via slack invitation (had some issues that @MichaelYochpaz mentioned had happened with other people when they tried to join), so we mainly communicated via comments on the PR or email (wolyslager@abnormalsecurity.com). I am free to schedule a Demo whenever you are 😄  I am on Central Time so anytime from 9am - 5pm CT works any day of the week. Let me know if that works for you, I can be more flexible if needed.

**Comment by wolyslager** [comment]
> > Hi @wolyslager, Thank you for your contribution! Good work :) We're ready for a demo. Please check [this page](https://xsoar.pan.dev/docs/contributing/demo-prep), and let me know when you're available for one over DFIR. Please fix the validation errors, you can see [here](https://app.circleci.com/pipelines/github/demisto/content/312108/workflows/60b1d55b-8e2d-43e1-9f35-0b26a8aa592c/jobs/726044) Feel free to reach out to me with any questions - I'm available here or on slack :)
> 
> Hey @sapirshuker! Thanks for taking a look 🙏 - I was unable to join via slack invitation (had some issues that @MichaelYochpaz mentioned had happened with other people when they tried to join), so we mainly communicated via comments on the PR or email ([wolyslager@abnormalsecurity.com](mailto:wolyslager@abnormalsecurity.com)). I am free to schedule a Demo whenever you are 😄 I am on Central Time so anytime from 9am - 5pm CT works any day of the week. Let me know if that works for you, I can be more flexible if needed.

@sapirshuker On second thought, there is some more functionality I need to add here (apologies) - I'll add that in and see if you have any comments!

**Comment by sapirshuker** [comment]
> > > Hi @wolyslager, Thank you for your contribution! Good work :) We're ready for a demo. Please check [this page](https://xsoar.pan.dev/docs/contributing/demo-prep), and let me know when you're available for one over DFIR. Please fix the validation errors, you can see [here](https://app.circleci.com/pipelines/github/demisto/content/312108/workflows/60b1d55b-8e2d-43e1-9f35-0b26a8aa592c/jobs/726044) Feel free to reach out to me with any questions - I'm available here or on slack :)
> > 
> > 
> > Hey @sapirshuker! Thanks for taking a look 🙏 - I was unable to join via slack invitation (had some issues that @MichaelYochpaz mentioned had happened with other people when they tried to join), so we mainly communicated via comments on the PR or email ([wolyslager@abnormalsecurity.com](mailto:wolyslager@abnormalsecurity.com)). I am free to schedule a Demo whenever you are 😄 I am on Central Time so anytime from 9am - 5pm CT works any day of the week. Let me know if that works for you, I can be more flexible if needed.
> 
> @sapirshuker On second thought, there is some more functionality I need to add here (apologies) - I'll add that in and see if you have any comments!

Hey, @wolyslager 
Thanks, Please let me know when the PR is ready for review. 
In addition, I sent to you an email with my details.
Please feel free to reach out to me with any questions - I'm available here or on email :)

**Comment by sapirshuker** [comment]
Hey, @wolyslager 
we haven’t heard from you in a while.
Do you need any help with the pull request?

Please feel free to reach out to me here or on Slack.
Thanks again for contributing to our repo, hope to hear from you soon.

**Comment by sapirshuker** [comment]
Hey, @wolyslager
we haven’t heard from you in a while.
Do you need any help with the pull request?

Please feel free to reach out to me here or on Slack.
Thanks again for contributing to our repo, hope to hear from you soon.

File: Packs/AbnormalSecurity/ReleaseNotes/2_1_2.md [hunk 1]
```diff
@@ -1,3 +1,3 @@
 #### Integrations
 ##### Abnormal Security
-- Updated the Docker image to: *demisto/python3:3.10.13.80014*.
+- Updated the Docker image to: *demisto/python3:3.10.13.80014*.
```
Comment (wolyslager) [inline Packs/AbnormalSecurity/ReleaseNotes/2_1_2.md:3]: not sure whats going on here? seems there is no change

**Comment by wolyslager** [comment]
Hey @sapirshuker not exactly sure what to do about this validation error 
```
[2023-11-22 04:52:25,400] [ERROR] /home/circleci/project/Packs/AbnormalSecurity/IncidentTypes/Abnormal_Security_Custom_Incident_types.json: [GR100] - Content item 'AbnormalSecurity' can be used in the 'marketplacev2, xsoar, xsoar_saas' marketplaces, however it uses content items: 'playbook0, playbook0, playbook0' which are not supported in all of the marketplaces of 'AbnormalSecurity'.
```

any ideas?

File: Packs/AbnormalSecurity/ReleaseNotes/2_1_2.md (hunk 1 above)
Comment (sapirshuker) [reply Packs/AbnormalSecurity/ReleaseNotes/2_1_2.md:3]: You need to add new Release Notes, you edited the `Packs/AbnormalSecurity/ReleaseNotes/2_1_2.md`
file that already exist.

**Comment by sapirshuker** [comment]
> Hey @sapirshuker not exactly sure what to do about this validation error
> 
> ```
> [2023-11-22 04:52:25,400] [ERROR] /home/circleci/project/Packs/AbnormalSecurity/IncidentTypes/Abnormal_Security_Custom_Incident_types.json: [GR100] - Content item 'AbnormalSecurity' can be used in the 'marketplacev2, xsoar, xsoar_saas' marketplaces, however it uses content items: 'playbook0, playbook0, playbook0' which are not supported in all of the marketplaces of 'AbnormalSecurity'.
> ```
> 
> any ideas?

@wolyslager 
you add the "playbookId": "playbook0" field to the incident type, do you need it? this incident type will trigger a playbook0.

**Review by melamedbn — APPROVED** [review]

File: Packs/AbnormalSecurity/ReleaseNotes/2_1_2.md (hunk 1 above)
Comment (wolyslager) [reply Packs/AbnormalSecurity/ReleaseNotes/2_1_2.md:3]: Yeah I just can't revert it, if you see the diff there is actually no change 

- Updated the Docker image to: *demisto/python3:3.10.13.80014*.
- Updated the Docker image to: *demisto/python3:3.10.13.80014*.

its the exact same line 🤔

Comment (wolyslager) [reply Packs/AbnormalSecurity/ReleaseNotes/2_1_2.md:3]: can't figure out how to do this cleanly - can we keep this change given its not actually changing anything @sapirshuker ?

**Comment by wolyslager** [comment]
@sapirshuker went ahead and removed the default playbook - let me know if any other changes are needed here!

File: Packs/AbnormalSecurity/Integrations/AbnormalSecurity/AbnormalSecurity.py [hunk 2]
```diff
@@ -730,29 +730,53 @@ def get_a_list_of_unanalyzed_abuse_mailbox_campaigns_command(client, args):
...
 
-def generate_threat_incidents(threats, current_iso_format_time):
+def generate_threat_incidents(client, threats, current_iso_format_time):
     incidents = []
     for threat in threats:
-        incident = {"dbotMirrorId": str(threat["threatId"]), "name": "Threat", "occurred": current_iso_format_time,
-                    'details': "Threat"}
+        threat_details = None
+        try:
+            threat_details = client.get_details_of_a_threat_request(threat["threatId"])
```
Comment (sapirshuker) [inline Packs/AbnormalSecurity/Integrations/AbnormalSecurity/AbnormalSecurity.py:738]: Do you not have all the details in the threat? Why do we need this request?

File: Packs/AbnormalSecurity/Integrations/AbnormalSecurity/AbnormalSecurity.py [hunk 3]
```diff
@@ -730,29 +730,53 @@ def get_a_list_of_unanalyzed_abuse_mailbox_campaigns_command(client, args):
...
     return incidents
 
 
-def generate_account_takeover_cases_incidents(cases, current_iso_format_time):
+def generate_account_takeover_cases_incidents(client, cases, current_iso_format_time):
     incidents = []
     for case in cases:
+        case_details = None
+        try:
+            case_details = client.get_details_of_an_abnormal_case_request(case["caseId"])
```
Comment (sapirshuker) [inline Packs/AbnormalSecurity/Integrations/AbnormalSecurity/AbnormalSecurity.py:774]: Do you not have all the details in the case? Why do we need this request?

**Review by sapirshuker — COMMENTED** [review]
Good Job!!!
Just few small comments and we will be ready for a demo.

File: Packs/AbnormalSecurity/Integrations/AbnormalSecurity/AbnormalSecurity.py (hunk 2 above)
Comment (wolyslager) [reply Packs/AbnormalSecurity/Integrations/AbnormalSecurity/AbnormalSecurity.py:738]: No- the request to /threats only fetches the new threat IDs - we then have to fetch the details - same goes for cases and campaigns

File: Packs/AbnormalSecurity/Integrations/AbnormalSecurity/AbnormalSecurity.py [hunk 4]
```diff
@@ -789,23 +813,23 @@ def fetch_incidents(
         all_incidents = []
 
         if fetch_threats:
-            threats_filter = f"receivedTime gte {last_fetch}"
+            threats_filter = f"receivedTime gte {last_fetch} lte 2023-06-30T01:01:01Z"
```
[the remark this answers is not in the document]
Comment (wolyslager) [reply Packs/AbnormalSecurity/Integrations/AbnormalSecurity/AbnormalSecurity.py:816]: ah yes - this is not supposed to be there

File: Packs/AbnormalSecurity/ReleaseNotes/2_2_0.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+##### Abnormal Security
+
+- Updated the Docker image to: *demisto/python3:3.10.13.80593*.
+
+#### Mappers
+
+##### New: Abnormal Security - Incoming Mapper
+
+- New: Abnormal Security Mapper (Available from Cortex XSOAR 6.9.0).
```
[the remark this answers is not in the document]
Comment (wolyslager) [reply Packs/AbnormalSecurity/ReleaseNotes/2_2_0.md:76]: Hey @sapirshuker I see the docker image update a few lines above (line 70) is there something wrong with that? 

https://github.com/demisto/content/pull/30393/files#diff-fcc1c5864a73c0534b08a06c1f2f0e6adff68f9024e3490f1d2452324f0b2462R70

File: Packs/AbnormalSecurity/Integrations/AbnormalSecurity/AbnormalSecurity.py [hunk 6]
```diff
@@ -730,29 +730,53 @@ def get_a_list_of_unanalyzed_abuse_mailbox_campaigns_command(client, args):
...
     return incidents
 
 
-def generate_abuse_campaign_incidents(campaigns, current_iso_format_time):
+def generate_abuse_campaign_incidents(client, campaigns, current_iso_format_time):
     incidents = []
     for campaign in campaigns:
+        campaign_details = None
+        try:
+            campaign_details = client.get_details_of_an_abuse_mailbox_campaign_request(campaign["campaignId"])
```
[the remark this answers is not in the document]
Comment (wolyslager) [reply Packs/AbnormalSecurity/Integrations/AbnormalSecurity/AbnormalSecurity.py:758]: same as above

File: Packs/AbnormalSecurity/Integrations/AbnormalSecurity/AbnormalSecurity.py (hunk 3 above)
Comment (wolyslager) [reply Packs/AbnormalSecurity/Integrations/AbnormalSecurity/AbnormalSecurity.py:774]: same as above

**Review by sapirshuker — COMMENTED** [review]
Hi @wolyslager, the code looks good!

We're ready for a demo. Please check [this page](https://xsoar.pan.dev/docs/contributing/demo-prep), and let me know when you're available for one over DFIR.

**Review by sapirshuker — APPROVED** [review]
Great job!!!

**Pull request merged**